### PR TITLE
ci: checking `Tui` prefix as required for exported interfaces

### PR DESCRIPTION
**Pull request by splincode**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Refactoring
- [ ] Code style update
- [x] Build or CI related changes
- [ ] Documentation content changes

### CODE REVIEW

File: projects/cdk/tokens/is-cypress.ts [hunk 1]
```diff
@@ -1,16 +1,22 @@
...
+export interface TuiWindowWithCypress extends WindowWithCypress {}
+
 /**
  * {@link https://docs.cypress.io/faq/questions/using-cypress-faq#Is-there-any-way-to-detect-if-my-app-is-running-under-Cypress Cypress docs}
  */
 export const TUI_IS_CYPRESS = new InjectionToken<boolean>(
     'Detect if app is running under Cypress',
     {
-        factory: () => !!inject<WindowWithCypress>(WINDOW).Cypress,
+        factory: () => !!inject<TuiWindowWithCypress>(WINDOW).Cypress,
```
Comment (waterplea) [inline projects/cdk/tokens/is-cypress.ts:20]: I don't think this adds anything to the type-safety. We might just type it `any` as well. This way we just pollute exports.

**Review by waterplea — APPROVED** [review]

File: projects/cdk/tokens/is-cypress.ts (hunk 1 above)
Comment (splincode) [reply projects/cdk/tokens/is-cypress.ts]: fixed

File: projects/i18n/interfaces/language.ts [hunk 2]
```diff
@@ -172,10 +206,18 @@ export type LanguagePreview = {
     };
 };
 
+export interface TuiLanguagePreview extends LanguagePreview {}
+
+/**
+ * @deprecated use {@link TuiLanguage}
+ */
+// eslint-disable-next-line @typescript-eslint/naming-convention
 export interface Language
```
Comment (nsbarsukov) [inline projects/i18n/interfaces/language.ts:215]: add migration to
https://github.com/Tinkoff/taiga-ui/blob/d8a6bd53f197a0379f9016abc3d3253e0f54e02a/projects/cdk/schematics/ng-update/constants/types.ts#L1-L7

File: projects/i18n/interfaces/language.ts [hunk 3]
```diff
@@ -159,8 +187,14 @@ export type LanguageEditor = {
...
-};
+}
+
+export interface TuiLanguageEditor extends LanguageEditor {}
 
+/**
+ * @deprecated use {@link TuiLanguagePreview}
+ */
+// eslint-disable-next-line @typescript-eslint/naming-convention
 export type LanguagePreview = {
```
Comment (nsbarsukov) [inline projects/i18n/interfaces/language.ts:198]: add migration

https://github.com/Tinkoff/taiga-ui/blob/d8a6bd53f197a0379f9016abc3d3253e0f54e02a/projects/cdk/schematics/ng-update/constants/types.ts#L1-L7

File: projects/i18n/interfaces/language.ts [hunk 4]
```diff
@@ -109,7 +131,13 @@ export interface LanguageTable {
     };
 }
 
-export type LanguageEditor = {
+export interface TuiLanguageTable extends LanguageTable {}
+
+/**
+ * @deprecated use {@link TuiLanguageEditor}
```
Comment (nsbarsukov) [inline projects/i18n/interfaces/language.ts:137]: add migration

File: projects/i18n/interfaces/language.ts [hunk 5]
```diff
@@ -84,6 +94,12 @@ export interface LanguageKit {
     };
 }
 
+export interface TuiLanguageKit extends LanguageKit {}
+
+/**
+ * @deprecated use {@link TuiLanguageCommerce}
```
Comment (nsbarsukov) [inline projects/i18n/interfaces/language.ts:100]: migration

File: projects/i18n/interfaces/language.ts [hunk 6]
```diff
@@ -20,6 +24,12 @@ export interface LanguageCore {
     countries: Record<TuiCountryIsoCode, string>;
 }
 
+export interface TuiLanguageCore extends LanguageCore {}
+
+/**
+ * @deprecated use {@link TuiLanguageKit}
```
Comment (nsbarsukov) [inline projects/i18n/interfaces/language.ts:30]: migration

File: projects/i18n/interfaces/language.ts [hunk 7]
```diff
@@ -3,6 +3,10 @@ import {TuiCountryIsoCode} from '@taiga-ui/i18n/enums';
 // prettier-ignore
 type MONTHS_ARRAY = [string, string, string, string, string, string, string, string, string, string, string, string];
 
+/**
+ * @deprecated use {@link TuiLanguageCore}
```
Comment (nsbarsukov) [inline projects/i18n/interfaces/language.ts:7]: migration

File: projects/demo/src/modules/pipes/filter/examples/1/component.ts [hunk 8]
```diff
@@ -29,5 +29,5 @@ export class TuiFilterExample1 {
         },
     ];
 
-    readonly matcher = (item: Item, search: number): boolean => item.price > search;
+    readonly matcher = (item: DemoItem, search: number): boolean => item.price > search;
```
Comment (nsbarsukov) [inline projects/demo/src/modules/pipes/filter/examples/1/component.ts:32]: can we just exclude `demo`-app from this rule ?

Comment (nsbarsukov) [reply projects/demo/src/modules/pipes/filter/examples/1/component.ts:32]: all demo-examples should be as simple as possible.

I am not sure that this prefix `Demo` makes example better.

File: projects/i18n/interfaces/language.ts (hunk 7 above)
Comment (splincode) [reply projects/i18n/interfaces/language.ts:7]: fixed

File: projects/i18n/interfaces/language.ts (hunk 5 above)
Comment (splincode) [reply projects/i18n/interfaces/language.ts:100]: fixed

File: projects/i18n/interfaces/language.ts [hunk 9]
```diff
@@ -97,6 +113,12 @@ export interface LanguageCommerce {
     cardExpiry: [string, string];
 }
 
+export interface TuiLanguageCommerce extends LanguageCommerce {}
+
+/**
+ * @deprecated use {@link TuiLanguageTable}
```
[the remark this answers is not in the document]
Comment (splincode) [reply projects/i18n/interfaces/language.ts:119]: fixed

File: projects/i18n/interfaces/language.ts (hunk 4 above)
Comment (splincode) [reply projects/i18n/interfaces/language.ts:137]: fixed

File: projects/i18n/interfaces/language.ts (hunk 2 above)
Comment (splincode) [reply projects/i18n/interfaces/language.ts:215]: fuxed

**Review by nsbarsukov — APPROVED** [review]

**Pull request merged**